facilitator/src/bin/facilitator.rs

```diff
@@ -717,22 +902,76 @@ fn batch_signing_key_from_arg(matches: &ArgMatches) -> Result<BatchSigningKey> {
     })
 }
 
-fn transport_for_output_path(
-    path: &str,
+fn ingestion_transport_and_keys_from_args(
+    matches: &ArgMatches,
+) -> Result<(
+    Box<dyn Transport>,
+    HashMap<String, UnparsedPublicKey<Vec<u8>>>,
```

Comment:
I think we need a couple levels of typedefs to make this and other spots more readable:

```
type Pubkeys = HashMap<String, UnparsedPublicKey<Vec<u8>>>;

struct VerifiableTransport(Box<dyn Transport>, Pubkeys);
```

I think we can then even do:

```
impl VerifiableTransport {
  from_args(matches: &ArgMatches) -> Self {
     ...
  }
}
```

Which isn't necessarily any less typing but conveys the idea a little more clearly of "here's how you make an X from a Y."

Reply:
I thought about this, but worried it'd make the diff even bigger than it is. I'll give it a try though, this kind of tidying can be soothing.